### DISCUSSION TITLE
HDD_SPARSIFY_ON_WRITE Performance impact?

### DISCUSSION THREAD

**Discussion by Blackpaw** [moosefs/moosefs#428] [Q&A]
I actually thought this defaulted to 0 (off) but from the chunkserver config file comments it's 1?

All my content is media files, which are unlikely to have any sparse holes in them. Any downsides to disabling HDD_SPARSIFY_ON_WRITE for chunkservers? I presume it will improve write performance.

**Comment by chogata** [comment]
Performance improvement is negligible - when you switch it off, you save like 2 processor instructions per one chunk replication, something you won't even notice with modern processors. A bit more when creating new chunks in the system, but still not much. But nothing bad will happen either, of course, so if your files are unlikely to be sparse, you can safely switch it off. OTOH if you do happen to have a sparse file, each replication will needlessly send zeros via network, then save them to disk, which takes time and occupies space. That's actually the reason why 1 (on) is default.

**Reply by Blackpaw** [reply]
Thanks, I'll leave it off then. I prefer to keep tweaks to a minimum these days.
